### ISSUE TITLE
Crash when changing the amount of instances in a `MultiMesh`

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#32500]
Windows 10 64 bits
```
Godot Engine v3.2.alpha.custom_build.24e1039eb - https://godotengine.org
Using GLES3 video driver
OpenGL ES 3.0 Renderer: GeForce GTX 1060 6GB/PCIe/SSE2
```
Happens too in 3.1.1.

I added an option in my plugin to change grass density, which involves changing the number of instances within a `MultiMesh`. It appears to work fine without errors, even if I momentarily set its value to zero and later non-zero.
However, it has a tendency to crash the editor. It happens after a few tries.

So I tried reproducing using my build of Godot master using a debugger, and that's what I could find:

![image](https://user-images.githubusercontent.com/1311555/66072801-a7aebd00-e54d-11e9-9776-a8a69713bc97.png)

Likely the problem is at the line with `glDrawElementsInstanced`. `count` is 2048, which I assume is triangles (although my mesh is a quad!), but I think it was supposed to be a different amount.

Eventually those errors show up:
```
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1282        Severity: High  Message: GL_INVALID_OPERATION error generated. Buffer name does not refer to an buffer object generated by OpenGL.
   At: drivers\gles3\rasterizer_gles3.cpp:123
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1282        Severity: High  Message: GL_INVALID_OPERATION error generated. Target buffer must be bound and not overlapped with mapping range.
   At: drivers\gles3\rasterizer_gles3.cpp:123
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1282        Severity: High  Message: GL_INVALID_OPERATION error generated. Buffer name does not refer to an buffer object generated by OpenGL.
   At: drivers\gles3\rasterizer_gles3.cpp:123
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1282        Severity: High  Message: GL_INVALID_OPERATION error generated. Invalid VAO/VBO/pointer usage.
   At: drivers\gles3\rasterizer_gles3.cpp:123
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1282        Severity: High  Message: GL_INVALID_OPERATION error generated. Invalid VAO/VBO/pointer usage.
   At: drivers\gles3\rasterizer_gles3.cpp:123
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL        Type: Error     ID: 1282        Severity: High  Message: GL_INVALID_OPERATION error generated. Invalid VAO/VBO/pointer usage.
   At: drivers\gles3\rasterizer_gles3.cpp:123
```

Here is a reproduction project in which I managed to reproduce it (two tries, two solid crashes):
[MultiMeshInstanceCountCrash.zip](https://github.com/godotengine/godot/files/3683125/MultiMeshInstanceCountCrash.zip)
Just run `main.tscn`.

-------
I previously couldn't find the correct combination of events that would reproduce the crash in a simple project, so I had also provided the original repro as it happened in my plugin:

[hterrain - Copy.zip](https://github.com/godotengine/godot/files/3683083/hterrain.-.Copy.zip)
1) Open `tests/test1/test1.tscn` (or execute Godot with `-v tests\test1\test1.tscn -e`)
2) Click on the terrain node
3) Add a `HTerrainDetailLayer` child node and select it
4) Get closer to the terrain, about that distance:
![image](https://user-images.githubusercontent.com/1311555/66077099-2825eb80-e557-11e9-8012-c57e5de1998f.png)
5) Paint some grass
![image](https://user-images.githubusercontent.com/1311555/66077125-370c9e00-e557-11e9-8453-3149b3f7f231.png)
6) Change `density` in the inspector to `0`, then back to any positive number, and keep changing it randomly using the arrows, setting it back to `0` or entering random numbers. The editor eventually crashes. If not, close Godot and restart to step 1).

The script in charge of the multimesh is `hterrain_detail_layer.gd`.

**Comment by mschulzeLpz** [comment]
Hi @Zylann I have also problems with multimesh when i set the instance_count multible times on the same multi mesh instance.
I download you test projekt and it crashes, but when i chaged your code to only call one time the *mm.instance_count = count* than it runs without errors.

I guess the Bug is located near the setter of *MultiMesh.instance_count*

**Comment by MadEqua** [comment]
This was also fixed by #34655 

Crashes consistently before the merge and runs for long minutes after the merge.
